### PR TITLE
Get all network devices by classification

### PR DESCRIPTION
**Pull request by cahmadis**
*  Added method to retrieve all network devices by classification.
*  Include category into MetasysObject model to get spaces.
*  Retrieve all child objects by Guid and object type.
*  Added EquipmentDefinitionName in MetasysObject
* Added ClassLevels and Description into AuditFilter and Audit respectively

@Federico-Artioli @ctumeyc  @jsharmbh Please review

### CODE REVIEW

File: MetasysServices/NetworkDevices/NetworkDeviceServiceProvider.cs [hunk 1]
```diff
@@ -58,7 +58,17 @@ public async Task<IEnumerable<MetasysObject>> GetAsync(string type = null)
             var response = await this.GetAllAvailablePagesAsync("networkDevices", new Dictionary<string, string> { { "type", type } }).ConfigureAwait(false);
             return ToMetasysObject(response, Version, MetasysObjectTypeEnum.Object);
         }
-
+        /// <inheritdoc/>
+        public IEnumerable<MetasysObject> GetByClassification(string classification = null)
+        {
+            return GetByClassificationAsync(classification).GetAwaiter().GetResult();
```
Comment (jsharmbh) [inline MetasysServices/NetworkDevices/NetworkDeviceServiceProvider.cs:64]: try to use sync method of `GetAllAvailablePagesAsync()` if the synchronous method is not available, then we can remove this method as we don't want to use GetAwaiter().GetResult() to make the 'async' method 'sync'.

we can remove the 'sync' variation of this method and allow the client/consumers to use only the 'async' method.

Just for reference, https://github.jci.com/PuneWorkflow/metasys-scan-engine/pull/49#issuecomment-214703 from Carol, that says we should not use as well as not create methods that are using .GetResult(), .wait().

File: MetasysServices/Interfaces/IMetasysClient.cs [hunk 2]
```diff
@@ -265,6 +265,16 @@ public interface IMetasysClient :IBasicService
         /// <inheritdoc cref="IMetasysClient.GetNetworkDevices(string)"/>
         Task<IEnumerable<MetasysObject>> GetNetworkDevicesAsync(string type = null);
 
+        /// <summary>
+        /// Gets all network devices by classification.
+        /// </summary>
+        /// <param name="classification">Optional classification as a string</param>
+        /// <exception cref="MetasysHttpException"></exception>
+        /// <exception cref="MetasysHttpParsingException"></exception>
+        IEnumerable<MetasysObject> GetNetworkDevicesByClassification(string classification = null);
```
Comment (Federico-Artioli) [inline MetasysServices/Interfaces/IMetasysClient.cs]: Actually I would deprecate the methods related to the Network Devices directly defined in the 'MetasysClient'.
(I would add it to the 'NetworkDevices' service)

File: MetasysServices/MetasysClient.cs [hunk 3]
```diff
@@ -473,6 +473,17 @@ public async Task<IEnumerable<MetasysObject>> GetNetworkDevicesAsync(string type
             return await NetworkDevices.GetAsync(type);
         }
 
+        /// <inheritdoc/>
+        public IEnumerable<MetasysObject> GetNetworkDevicesByClassification(string classification = null)
```
Comment (Federico-Artioli) [inline MetasysServices/MetasysClient.cs]: Actually I would deprecate the methods related to the Network Devices directly defined in the 'MetasysClient'.
(I would add it to the 'NetworkDevices' service)

File: MetasysServices/NetworkDevices/NetworkDeviceServiceProvider.cs [hunk 4]
```diff
@@ -58,7 +58,17 @@ public async Task<IEnumerable<MetasysObject>> GetAsync(string type = null)
...
+        }
+        /// <inheritdoc/>
+        public async Task<IEnumerable<MetasysObject>> GetByClassificationAsync(string classification = null)
```
Comment (Federico-Artioli) [inline MetasysServices/NetworkDevices/NetworkDeviceServiceProvider.cs:62]: Would it be possible to define this method as overload of  the method 'Get()' (adding the classification parameter) ?
Also, what about define a specific 'enum' (e.g. 'NetworkDeviceClassificationEnum') to better specify the parameter 'classification' ?

File: MetasysServices/MetasysClient.cs (hunk 3 above)
Comment (cahmadis) [reply MetasysServices/MetasysClient.cs]: Removed method GetNetworkDevicesByClassification from 'MetasysClient' and added to 'NetworkDevices' service

File: MetasysServices/NetworkDevices/NetworkDeviceServiceProvider.cs (hunk 4 above)
Comment (cahmadis) [reply MetasysServices/NetworkDevices/NetworkDeviceServiceProvider.cs:68]: Added method GetAsync to INetworkDeviceService and NetworkDeviceServiceProvider.cs and created enum 'NetworkDeviceClassificationEnum'

**Pull request merged**